### ISSUE TITLE
rollout/ep_len_mean and rollout/ep_rew_mean not shown despite env wrapped in Monitor

### ISSUE THREAD

**Issue by Elcoid** [DLR-RM/stable-baselines3#1806]
### 🐛 Bug

Issues #232 and #251 describe the same problem as the one I have: the `rollout/ep_len_mean` and `rollout/ep_rew_mean` are not showed in the terminal or in Tensorboard. However the suggested solution of wrapping the environment in a Monitor does not work in my case. It used to work before (a few months ago, but I was running a quite old version then, I don't remember which one though), but I only noticed today that the two graphs were missing. The difference between before and now is probably due to updates. I do remember I had to adjust the code because the API changed. Also I had to use gymnasium instead of gym in some places.

Is there something else that has to be done other than wrapping in a Monitor in order to get `ep_len_mean` and `ep_rew_mean`?

I ran the `env_checker`. It complained about observations not being in the range of the observation space (`spaces.Box`). However, that part didn't change since before the graphs disappeared. So I don't think it's related to my problem.

### Code example

The code I'm using is pretty big and almost messy, so I will realistically not be able to provide a minimal working example. Here are relevant snippets:
```python
import numpy as np
import gym
from gymnasium import spaces
from stable_baselines3 import PPO
from stable_baselines3.ppo import MlpPolicy
from stable_baselines3.common.vec_env import DummyVecEnv, VecNormalize, VecFrameStack, SubprocVecEnv
from stable_baselines3.common.callbacks import BaseCallback, CheckpointCallback, CallbackList
from stable_baselines3.common.utils import set_random_seed
from stable_baselines3.common.monitor import Monitor

os.environ["CUDA_VISIBLE_DEVICES"] = ""

# ...

class CustomGym(gym.Env):
	def __init__(self):
		super(CustomGym, self).__init__()
		self.action_space = spaces.Box(
			low = -1,
			high = 1,
			shape = (3,),
			dtype = np.float32
		)
		self.observation_space = spaces.Box(
			low = -1,
			high = 1,
			shape = (43,),
			dtype = np.float32
		)
		self._obs = self.observation_space.sample()

		# ...

	def step(self, action):
		# ...
		return (self._obs, self._rew, terminated, False, {})

def make_env(rank, seed = 0):
	def _init():
		env = CustomGym()
		env.seed(seed + rank)
		env = Monitor(env, allow_early_resets = True)
		return env
	set_random_seed(seed)
	return _init

# Main part
env = SubprocVecEnv([make_env(i) for i in range(num_envs)]) # num_envs is defined somewhere else
model = PPO(
	MlpPolicy,
	env,
	gamma = 0.95,
	n_steps = 2048,
	batch_size = 1024,
	learning_rate = 0.001,
	tensorboard_log = "/some/path",
	policy_kwargs = {...} # A dict defined somewhere else,
	verbose = 1,
	n_epochs = 10
)

chkpt_callback = CheckpointCallback(
	save_freq = 1000,
	save_path = "/another/path",
	name_prefix = ""
)

# CallbackList because sometimes another callback is used,
# but now it is commented out
callback = CallbackList([chkpt_callback])

model.learn(
	total_timesteps = int(1e9),
	tb_log_name = "model_name",
	callback = callback,
	reset_num_timesteps = True
)
```
Please tell me if there are other parts that should be included.

### Relevant log output / Error message

```shell
------------------------------------------
| time/                   |              |
|    fps                  | 13           |
|    iterations           | 2            |
|    time_elapsed         | 1253         |
|    total_timesteps      | 16384        |
| train/                  |              |
|    approx_kl            | 0.0057509197 |
|    clip_fraction        | 0.0447       |
|    clip_range           | 0.2          |
|    entropy_loss         | -4.24        |
|    explained_variance   | -0.000419    |
|    learning_rate        | 0.001        |
|    loss                 | 1.41e+03     |
|    n_updates            | 10           |
|    policy_gradient_loss | -0.00569     |
|    std                  | 0.992        |
|    value_loss           | 6.94e+03     |
------------------------------------------
```

### System Info

The libraries were installed by `pip` in a Miniconda environment.

- OS: Linux-5.15.0-91-generic-x86_64-with-glibc2.31 # 101~20.04.1-Ubuntu SMP Thu Nov 16 14:22:28 UTC 2023
- Python: 3.11.5
- Stable-Baselines3: 2.1.0
- PyTorch: 2.1.0+cu121
- GPU Enabled: False
- Numpy: 1.26.1
- Cloudpickle: 3.0.0
- Gymnasium: 0.29.1
- OpenAI Gym: 0.26.2

Note: `get_system_info` complained that the Nvidia driver on my system is too old, but I don't use Cuda for this project.

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014) example to reproduce the bug
- [X] I have checked my env using the env checker
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by Elcoid** [comment]
Hello, as I was writing my issue above, I realized the problem could be coming from an incompatibility between `gym` and `gymnasium`. So I replaced `import gym` with `import gymnasium as gym` and made some more tests. However this does not solve the problem.

**Comment by araffin** [comment]
Hello,
if you cannot provide minimal and working example, we cannot really help.

The following work for me, maybe env never terminates?

```python
import gymnasium as gym
import numpy as np
from gymnasium import spaces
from stable_baselines3 import PPO
from stable_baselines3.common.env_checker import check_env
from stable_baselines3.common.env_util import make_vec_env
from stable_baselines3.common.vec_env import SubprocVecEnv


class CustomGym(gym.Env):
    def __init__(self):
        super().__init__()
        self.action_space = spaces.Box(low=-1, high=1, shape=(3,), dtype=np.float32)
        self.observation_space = spaces.Box(
            low=-1, high=1, shape=(43,), dtype=np.float32
        )
        self._obs = self.observation_space.sample()

    def reset(self, seed=None, options=None):
        return self._obs, {}

    def step(self, action):
        terminated = np.random.rand() > 0.8
        return self._obs, 1.0, terminated, False, {}


if __name__ == "__main__":
    num_envs = 2
    check_env(CustomGym())

    vec_env = make_vec_env(CustomGym, n_envs=num_envs, vec_env_cls=SubprocVecEnv)
    model = PPO("MlpPolicy", vec_env, n_steps=2048, verbose=1, n_epochs=1)

    model.learn(total_timesteps=1024)
```

**Comment by Elcoid** [comment]
Hello, thank you for your answer.

> maybe env never terminates?

Is this controlled by the `terminated` output of the `step` method? If so, there are times where it returns True and others where it returns False, so it's probably not the problem. If there is no obvious mistake and it's not a known bug, I will try to trim fluff off my code and isolate what's wrong.

**Comment by Elcoid** [comment]
Hello,
It turns out the case in `step` where `terminated` is returned as True never happens because of a recent change. So your intuition was right!

I'm curious about something however. In a test script that I wrote, the reward returned by `step` is always 1. However `ep_rew_mean` is always different. It seems to be very roughly equal to the inverse of the probability that `terminated` be true (i.e. `np.random.rand() > 0.9` ⇒ `ep_rew_mean` ≈ 10, `np.random.rand() > 0.999` ⇒ `ep_rew_mean` ≈ 1000, etc.). Do I misunderstand what `ep_rew_mean` is supposed to be, or there is a problem?

**Comment by araffin** [comment]
https://stable-baselines3.readthedocs.io/en/master/common/logger.html#explanation-of-logger-output

episodic reward should be called return, it is fine.
